crates/oxc_ast/src/ast/js.rs

```diff
@@ -1597,7 +1597,7 @@ pub struct Class<'a> {
     pub id: Option<BindingIdentifier<'a>>,
     #[scope(enter_before)]
     pub type_parameters: Option<Box<'a, TSTypeParameterDeclaration<'a>>>,
-    #[visit_as(ClassHeritage)]
+    #[visit(alias(ClassHeritage))]
```

Comment:
Using `alias` instead of `as` because `as` introduces an edge case since it is a keyword and not a `Path` item.

We are already dealing with such edge cases for `#[scope(if(...))]` but that one feels natural to the user, On the other hand `as(...)` doesn't.

Reply:
How much work would it be to enable `visit(as(...))` (rather than `alias`)?

If it's not too much work, I think `as` would be preferable. I know `as` has a specific meaning in Rust, but the meaning of the attr in English really is "visit this field *as* as a `ClassHeritage`", so "as" is I think the best word to use. "Alias" is a bit more mysterious.

As with `visit(args(...))`, I hope we can eventually standardize and remove this special attr by adding an actual `ClassHeritage` AST node type (or by not having a specific visitor for it, if there's some other way to fulfil the use case). But who knows when/if we'll get to that, so I think it'd be better to leave the code now in best state we can, rather than assuming it doesn't much matter because we'll be removing it anyway.

Reply:
> How much work would it be to enable visit(as(...)) (rather than alias)?

It is quite trivial, We try to parse it as the keyword and if it fails we proceed to other options. It would be quite similar to this(but a bit more complicated since it is nested):

https://github.com/oxc-project/oxc/blob/759180e95c1c23e432f3f41bf8c6a8245080c335/tasks/ast_codegen/src/markers.rs#L76-L82

> As with visit(args(...)), I hope we can eventually standardize and remove this special attr by adding an actual ClassHeritage AST node type (or by not having a specific visitor for it, if there's some other way to fulfil the use case). But who knows when/if we'll get to that, so I think it'd be better to leave the code now in best state we can, rather than assuming it doesn't much matter because we'll be removing it anyway.

I totally agree, I've introduced this just to avoid having edge cases/hacky workarounds in the generator.

------------

I'll request a review when I'm done refactoring it.

Reply:
Thanks Rez. I've reviewed the rest of PR, and no further comments other than this one (and my nits below).